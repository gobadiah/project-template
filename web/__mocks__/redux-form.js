const reduxForm = jest.genMockFromModule('redux-form');

module.exports = reduxForm;
