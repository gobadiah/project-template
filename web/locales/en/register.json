{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
