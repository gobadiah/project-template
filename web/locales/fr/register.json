{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
