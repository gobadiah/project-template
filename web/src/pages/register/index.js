import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
