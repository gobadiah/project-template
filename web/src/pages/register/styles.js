import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
