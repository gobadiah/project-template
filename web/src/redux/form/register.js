import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
