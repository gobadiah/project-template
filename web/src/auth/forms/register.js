import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
