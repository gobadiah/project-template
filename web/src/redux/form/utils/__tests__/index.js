import 'jsdom-global/register';

import React from 'react';
import { shallow } from 'enzyme';

import { SubmissionError } from 'redux-form';

import {
  formatError,
  formatWarning,
  handleFormErrors,
  BaseForm,
} from '..';

describe('formatError', () => {
  it('should format a string given the t function', () => {
    const t = jest.fn(x => x);
    const error = 'This is an error';
    expect(formatError(error, t)).toBe(error);
    expect(t).toHaveBeenCalledTimes(1);
    expect(t).toHaveBeenCalledWith(error);
  });

  it('should format an error of string given the t function', () => {
    const t = jest.fn(x => x);
    const error = ['one error', 'two error'];
    expect(formatError(error, t)).toEqual([
      error[0],
      error[1],
    ]);
    expect(t).toHaveBeenCalledTimes(2);
    expect(t).toHaveBeenCalledWith(error[0]);
    expect(t).toHaveBeenCalledWith(error[1]);
  });
});

describe('formatWarning', () => {
  it('should format a string given the t function', () => {
    const t = jest.fn(x => x);
    const error = 'This is an error';
    expect(formatWarning(error, t)).toBe(error);
    expect(t).toHaveBeenCalledTimes(1);
    expect(t).toHaveBeenCalledWith(error);
  });

  it('should format an error of string given the t function', () => {
    const t = jest.fn(x => x);
    const error = ['one error', 'two error'];
    expect(formatWarning(error, t)).toEqual([
      error[0],
      error[1],
    ]);
    expect(t).toHaveBeenCalledTimes(2);
    expect(t).toHaveBeenCalledWith(error[0]);
    expect(t).toHaveBeenCalledWith(error[1]);
  });
});

describe('handleFormErrors', () => {
  it('should rethrow if the error doesn\'t match the conditions', () => {
    const error = new Error();
    expect(() => handleFormErrors(error)).toThrowError(error);
  });

  it('should throw a SubmissionError if conditions match', () => {
    const error = {
      response: {
        status: 400,
        data: {
          non_field_errors: [
            'This is a global error',
          ],
          errors: {
            email: 'This is an email error',
          },
        },
      },
    };
    expect(() => handleFormErrors(error)).toThrow(SubmissionError);
  });
});

describe('BaseForm', () => {
  it('should be subclass and render an empty form', () => {
    const error = ['This is an error'];
    const warning = ['This is a warning'];
    const formatErr = err => `<error>${err}</error>`;
    const formatWarn = warn => `<warning>${warn}</warning>`;
    const handleSubmit = jest.fn();
    const pristine = false;
    const submitting = false;
    const reset = jest.fn();
    const t = x => x;
    const i18n = {
      on: jest.fn(),
    };

    const wrapper = shallow(<BaseForm
      error={error}
      warning={warning}
      formatError={formatErr}
      formatWarning={formatWarn}
      handleSubmit={handleSubmit}
      pristine={pristine}
      submitting={submitting}
      reset={reset}
    />, { context: { i18n, t } });
    expect(wrapper).toMatchSnapshot();
  });
});
