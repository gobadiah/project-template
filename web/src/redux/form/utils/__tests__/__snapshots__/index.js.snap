// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseForm should be subclass and render an empty form 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    disabled={false}
    type="submit"
    value="Send"
  />
  <input
    disabled={false}
    onClick={[MockFunction]}
    type="button"
    value="Reset"
  />
  &lt;warning&gt;This is a warning&lt;/warning&gt;
  &lt;error&gt;This is an error&lt;/error&gt;
</form>
`;
